cmd/agent/install_mac_os.sh

```diff
@@ -45,7 +45,12 @@ if [ ! $apikey ]; then
 fi
 
 # get real user (in case of sudo)
-real_user=`logname`
+real_user=`if [ $SUDO_USER ]; then
+  echo $SUDO_USER
+else
+  echo $USER
```

Comment:
do you have some supporting docs on what the `SUDO_USER` and `USER` env vars are supposed to contain, and explain what you're trying to do with this block?

Reply:
Seems like there are no official manpages repository for macOS available online, so I searched online for versions that correspond to the ones I have on my mac:

`SUDO_USER` is defined in `man sudo`: https://linux.die.net/man/8/sudo
`USER` is defined in `man login`: https://ss64.com/osx/login.html

As for what I'm trying to do with this block: we want to get the "real user" which launched the command. Two situations can happen:
- the command was run as the current user: then `$USER` contains the user which launched the command, and `$SUDO_USER` is empty,
- the command was run with `sudo`: then `$USER` contains the name of the user targeted by the `sudo` command (by default, `root`) and `$SUDO_USER` contains the user which launched the `sudo` command,.
In both cases, we get what we want: `real_user` contains the username which originally launched the command.

I'll add this as comments in the script.